### ISSUE TITLE
[QUERY] I keep hitting - Idle connection closed by service just after sending CBS token. Very rare case. Will retry.

### ISSUE THREAD

**Issue by chester-238** [Azure/azure-sdk-for-java#17994]
**Query/Question**

I have this SDK used in a Spring Boot app which runs a cron job every half hour. It seems to run fine initially then all of a sudden we start getting this error - **_Idle connection closed by service just after sending CBS token. Very rare case. Will retry._**

We are using version 3.4.0. The only recent change is the ASB and the Java app are now on different resource groups. 

Is there anything known that causes this issue?

***Why is this not a Bug or a feature Request?***

This was working fine until recently and requires a restart of the app which seems to work fine before throwing issues again.

**Setup (please complete the following information if applicable):**

 - Version of the Library used: 3.4.0
 - Unable to reproduce this locally but only on running container
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ X] Query Added
- [ X] Setup information Added

**Comment by joshfree** [comment]
@hemanttanwar can you please assist?

**Comment by hemanttanwar** [comment]
@chester-238 Are you using this library ? https://mvnrepository.com/artifact/com.microsoft.azure/azure-servicebus/3.4.0

**Comment by yvgopal** [comment]
I understand when this WARN is written to the logs. But I don't think it requires an application restart. The SDK will retry connection creation. Could you confirm if the application is not recovering at all? If so, could you please capture traces subsequent to this WARN trace and attach it to this issue?

Looks like you are creating a QueueClient/TopicClient/SubscriptionClient and keeping it in memory even though your job runs only once every 30 minutes. It shouldn't be problem, but better practice **in the case of such periodic jobs** is to create clients when needed.

**Comment by hemanttanwar** [comment]
@chester-238 I will close this issue. 
If you still have any issue, you can reopen this or create new issue if you need more information on this.
